### PR TITLE
MAJOR FIX: Fix Private Methods not getting mocked properly

### PR DESCRIPTION
**Pull request by TehWardy**
Note:

When all tests are run together the system runs in to a threading bug due to the static exception defined in InternalMockOrchestrationService

I tried to find ways to pass in "thread safe variations" but hit some weirdness.
I think we need to figure out how to pass the method itself down in to the Harmony lib then have that hold a reference to it to use or hold non static references higher up the application stack somewhere.

Bottom line ... it works but not multi threaded.
I'll think more on this one.

### CODE REVIEW

File: InternalMock.Tests.Unit/Services/Foundations/ExampleService.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
```
Comment (hassanhabib) [inline InternalMock.Tests.Unit/Services/Foundations/ExampleService.cs]: Add the header here.

**Comment by TehWardy** [comment]
image

No idea why the checks are failing here, this is what I see locally.

File: InternalMock.Tests.Unit/Services/Foundations/ExampleService.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        void TryCatch(VoidFunction voidFunction)
+        {
+            try
+            {
+                voidFunction();
+            }
+            catch (NullReferenceException exception)
+            {
+                // illustrative only
```
Comment (hassanhabib) [inline InternalMock.Tests.Unit/Services/Foundations/ExampleService.cs]: Remove comments

File: InternalMock.Tests.Unit/Services/Foundations/Patchings/PatchServiceTests.cs [hunk 3]
```diff
@@ -66,5 +67,22 @@ private static MethodInfo CreateRandomMethodInfo()
                 returnType: randomType,
                 parameterTypes: randomTypes);
         }
+
+        static Exception RandomException()
+            => new Exception []
```
Comment (hassanhabib) [inline InternalMock.Tests.Unit/Services/Foundations/Patchings/PatchServiceTests.cs:72]: Scope this. use `{}` instead of fat-arrow since it's a multi-liner

File: InternalMock.Tests.Unit/Services/Foundations/Reflections/ReflectionServiceTests.Logic.cs [hunk 4]
```diff
@@ -46,5 +48,26 @@ public void ShouldRetrieveMethodInfo()
...
         }
+
+        [Fact]
+        public void ShouldRetrievePrivateMethodInfo()
+        {
+            // given
+            var reflectionService = new ReflectionService(new ReflectionBroker());
+            var privateMethodName = "DoPrivateStuff";
+            Type targetType = typeof(ExampleService);
+            MethodInfo expectedMethodInfo = targetType.GetMethod(
```
Comment (hassanhabib) [inline InternalMock.Tests.Unit/Services/Foundations/Reflections/ReflectionServiceTests.Logic.cs:62]: Put a new line between multi-liners and single-liner

File: InternalMock.Tests.Unit/Services/Foundations/Reflections/ReflectionServiceTests.Logic.cs [hunk 5]
```diff
@@ -46,5 +48,26 @@ public void ShouldRetrieveMethodInfo()
 
             this.reflectionBrokerMock.VerifyNoOtherCalls();
         }
+
+        [Fact]
+        public void ShouldRetrievePrivateMethodInfo()
+        {
+            // given
+            var reflectionService = new ReflectionService(new ReflectionBroker());
+            var privateMethodName = "DoPrivateStuff";
```
Comment (hassanhabib) [inline InternalMock.Tests.Unit/Services/Foundations/Reflections/ReflectionServiceTests.Logic.cs]: Define explicit type here instead of `var`

File: InternalMock.Tests.Unit/Services/Foundations/Reflections/ReflectionServiceTests.Logic.cs [hunk 6]
```diff
@@ -46,5 +48,26 @@ public void ShouldRetrieveMethodInfo()
 
             this.reflectionBrokerMock.VerifyNoOtherCalls();
         }
+
+        [Fact]
+        public void ShouldRetrievePrivateMethodInfo()
+        {
+            // given
+            var reflectionService = new ReflectionService(new ReflectionBroker());
```
Comment (hassanhabib) [inline InternalMock.Tests.Unit/Services/Foundations/Reflections/ReflectionServiceTests.Logic.cs]: Define a variable for `ReflectionBroker` then pass it here
